[libpq/10.4] Can not found debug build libpq by CMake on Windows OS

CMake fails when running `find_library(PostgreSQL REQUIRED)` when using a debug-built libpq 10.4 on Windows OS.

The `FindPostgreSQL.cmake` provided by CMake looks for libpq.lib but what actually exists is libpqd.lib.

### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **libpq/10.4** (https://github.com/bincrafters/conan-libpq)
  * Operating System+version: **Windows 10 Pro 1809 (`ver` command output: "Microsoft Windows [Version 10.0.17763.557]")**
  * Compiler+version: **Visual Studio 2017 Professional**
  * Docker image: **None**
  * Conan version: **conan 1.16.1**
  * Python version: **None (conan installed by Windows installer)**

### Conan profile

```
[settings]
arch=x86_64
arch_build=x86_64
build_type=Debug
compiler=Visual Studio
compiler.runtime=MDd
compiler.version=15
os=Windows
os_build=Windows
[options]
[build_requires]
[env]
```

### Steps to reproduce (Include if Applicable)

I created small project: https://github.com/tt4g/reproduce-conan-libpq-not-found-by-cmake

Use the following `conanfile.txt`:

```
[requires]
libpq/10.4@bincrafters/stable

[options]
libpq:shared=False
libpq:with_zlib=False
libpq:with_openssl=False

[generators]
cmake
```

and the following `CMakeLists.txt`:

```
cmake_minimum_required(VERSION 3.0.0)
project(reproduce_conan_libpq_not_found_by_cmake VERSION 0.1.0)

include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
conan_basic_setup()

find_package(PostgreSQL REQUIRED)
```

run the following command:

```bash
mkdir build
cd build
conan install .. --build missing -s build_type=Debug
cmake -DCMAKE_BUILD_TYPE=Debug ..
```

### Logs (Include/Attach if Applicable)

```
$ mkdir build
$ cd build
$ conan install .. --build missing -s build_type=Debug
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Debug
compiler=Visual Studio
compiler.runtime=MDd
compiler.version=15
os=Windows
os_build=Windows
[options]
[build_requires]
[env]

libpq/10.4@bincrafters/stable: Not found in local cache, looking in remotes...
libpq/10.4@bincrafters/stable: Trying with 'conan-center'...
Downloading conanmanifest.txt
[==================================================] 166B/166B
Downloading conanfile.py
[==================================================] 5.2KB/5.2KB
Downloading conan_export.tgz
[==================================================] 764B/764B
Decompressing conan_export.tgz: 100%|██████████| 764/764 [00:00<00:00, 11.9kB/s]
libpq/10.4@bincrafters/stable: Downloaded recipe revision 0
conanfile.txt: Installing package
Requirements
    libpq/10.4@bincrafters/stable from 'conan-center' - Downloaded
Packages
    libpq/10.4@bincrafters/stable:8cf01e2f50fcd6b63525e70584df0326550364e1 - Download

libpq/10.4@bincrafters/stable: Retrieving package 8cf01e2f50fcd6b63525e70584df0326550364e1 from remote 'conan-center' 
Downloading conanmanifest.txt
[==================================================] 1.3KB/1.3KB
Downloading conaninfo.txt
[==================================================] 522B/522B
Downloading conan_package.tgz
[==================================================] 546.8KB/546.8KB
Decompressing conan_package.tgz: 100%|██████████| 534k/534k [00:00<00:00, 2.58MB/s]
libpq/10.4@bincrafters/stable: Package installed 8cf01e2f50fcd6b63525e70584df0326550364e1
libpq/10.4@bincrafters/stable: Downloaded package revision 0
conanfile.txt: Generator cmake created conanbuildinfo.cmake
conanfile.txt: Generator txt created conanbuildinfo.txt
conanfile.txt: Generated conaninfo.txt
conanfile.txt: Generated graphinfo

$ cmake -DCMAKE_BUILD_TYPE=Debug ..
-- Building for: Visual Studio 15 2017
-- The C compiler identification is MSVC 19.16.27031.1
-- The CXX compiler identification is MSVC 19.16.27031.1
-- Check for working C compiler: <MSVC_INSTALL_PATH>/Microsoft Visual Studio/2017/Professional/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x86/cl.exe
-- Check for working C compiler: <MSVC_INSTALL_PATH>/Microsoft Visual Studio/2017/Professional/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x86/cl.exe -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: <MSVC_INSTALL_PATH>/Microsoft Visual Studio/2017/Professional/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x86/cl.exe
-- Check for working CXX compiler: <MSVC_INSTALL_PATH>/Microsoft Visual Studio/2017/Professional/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x86/cl.exe -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Conan: Adjusting output directories
-- Conan: Using cmake global configuration
-- Conan: Adjusting default RPATHs Conan policies
-- Conan: Adjusting language standard
-- Current conanbuildinfo.cmake directory: <WORKSPACE>/reproduce-conan-libpq-not-found-by-cmake/build
CMake Error at <CMAKE_INSTALL_PATH>/cmake-3.14/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find PostgreSQL (missing: PostgreSQL_LIBRARY) (found version
  "10.4")
Call Stack (most recent call first):
  <CMAKE_INSTALL_PATH>/cmake-3.14/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  <CMAKE_INSTALL_PATH>/cmake-3.14/Modules/FindPostgreSQL.cmake:211 (find_package_handle_standard_args)
  CMakeLists.txt:7 (find_package)


-- Configuring incomplete, errors occurred!
See also "<WORKSPACE>/reproduce-conan-libpq-not-found-by-cmake/build/CMakeFiles/CMakeOutput.log".
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -17,7 +17,6 @@ endif()
 
 set(CMAKE_STATIC_LIBRARY_PREFIX)
 set(CMAKE_SHARED_LIBRARY_PREFIX)
-set(CMAKE_DEBUG_POSTFIX "d")
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
 
 file(WRITE source_subfolder/src/port/pg_config_paths.h "#define PGBINDIR \"${CONAN_CMAKE_MODULE_PATH}/bin\"\n")
@@ -105,7 +104,7 @@ endif()
 include_directories(source_subfolder/src/include/port/win32 source_subfolder/src/include/port/win32_msvc)
 
 set(pg_backend_src
-    source_subfolder/src/common/base64.c    
+    source_subfolder/src/common/base64.c
     source_subfolder/src/common/config_info.c
     source_subfolder/src/common/controldata_utils.c
     source_subfolder/src/common/fe_memutils.c
@@ -120,7 +119,7 @@ set(pg_backend_src
     source_subfolder/src/common/rmtree.c
     source_subfolder/src/common/saslprep.c
     source_subfolder/src/common/scram-common.c
-    source_subfolder/src/common/sha2.c    
+    source_subfolder/src/common/sha2.c
     source_subfolder/src/common/unicode_norm.c
     source_subfolder/src/common/username.c
     source_subfolder/src/common/wait_error.c
@@ -138,7 +137,7 @@ set(pg_libpq_src
     source_subfolder/src/interfaces/libpq/fe-print.c
     source_subfolder/src/interfaces/libpq/fe-protocol2.c
     source_subfolder/src/interfaces/libpq/fe-protocol3.c
-    source_subfolder/src/interfaces/libpq/fe-secure.c    
+    source_subfolder/src/interfaces/libpq/fe-secure.c
     source_subfolder/src/interfaces/libpq/libpq-events.c
     source_subfolder/src/interfaces/libpq/pqexpbuffer.c
     source_subfolder/src/interfaces/libpq/libpq-dist.rc

